apps/portal/prisma/new-schema.prisma

```diff
@@ -0,0 +1,229 @@
+    userId       String
+    expires      DateTime
+    user         User     @relation(fields: [userId], references: [id], onDelete: Cascade)
+
+    @@map("sessions")
+}
+
+model Role {
+    id       String @id @default(auto()) @map("_id") @db.ObjectId
+    roleName String
+
+    RolesOnUser RolesOnUser[]
+
+    @@map("roles")
+}
+
+model RolesOnUser {
+    id        String   @id @default(auto()) @map("_id") @db.ObjectId
+    role      Role     @relation(fields: [roleId], references: [id])
+    roleId    String
+    user      User     @relation(fields: [userId], references: [id])
+    userId    String
+    createdAt DateTime @default(now())
+
+    @@unique([userId, roleId])
+}
+
+model User {
+    id            String        @id @default(auto()) @map("_id") @db.ObjectId
+    name          String?
+    email         String?       @unique
+    emailVerified DateTime?
+    image         String?
+    accounts      Account[]
+    sessions      Session[]
+    profile       Profile?
+    roles         RolesOnUser[]
+
+    applications    UserAppJunction[]   @relation("UserLink")
+
+    @@map("users")
+}
+
+model VerificationToken {
+    id         String   @id @default(auto()) @map("_id") @db.ObjectId
+    identifier String
+    token      String   @unique
+    expires    DateTime
+
+    @@unique([identifier, token])
+}
+
+model Profile {
+    id               String    @id @default(auto()) @map("_id") @db.ObjectId
+    firstName        String
+    lastName         String
+    email            String    @unique
+    netid            String    @unique
+    classStanding    String
+    major            String
+    utdStudent       Boolean
+    roles            String[]
+    membershipStatus Boolean
+    membershipTS     DateTime?
+    resume           Boolean
+    resumeTS         DateTime?
+
+    user User @relation(fields: [userId], references: [id])
+
+    userId String             @unique @db.ObjectId
+    events EventReservation[]
+
+    officerId   String             @unique @db.ObjectId
+    officer     Officer            @relation(fields: [officerId], references: [id])
+}
+
+model Event {
+    id          String             @id @default(auto()) @map("_id") @db.ObjectId
+    summary     String
+    description String
+    url         String
+    location    String
+    start       DateTime
+    end         DateTime
+    profiles    EventReservation[]
+
+    @@map("events")
+}
+
+model TypeformApplication {
+    id                  String  @id @default(auto()) @map("_id") @db.ObjectId
+    active              Boolean
+    description         String
+    endpoint            String
+    externalResourceUrl String
+    typeformId          String
+    typeformName        String
+
+    @@map("typeform_applications")
+}
+
+model Application {
+    id                  String  @id @default(auto()) @map("_id") @db.ObjectId
+    createdAt           DateTime
+    expireDate          DateTime
+    type                String?
+    role                String?
```

Comment:
I was actually going to ask you about that. 

In the doc, it is defined as Role[] for the application but I'm unsure what that entails for an application?

Reply:
done w the rest of changes - i just need your clarification on this please

Reply:
The `Role` object is for role-based access control. Currently, `officer` is the only role and it is used to ensure that only officers will have access to administrative features such as create events, applications, and Vanity link, and the plan is so that we can add another role like `sponsor` since there are plans to create features for sponsors to download resumes. Since it seems like applications are managed by division, perhaps it would be better to use `division` (or `department`) to control who has write permission to the application.

Reply:
what is the difference between Role and RolesOnUser?

I'm thinking to just connect to Role with a 1-to-1

As per your division, I'll connect it to Division with a 1-to-1

What do you think?

Reply:
@DedsecKnight - can i get a resolve on this?

Reply:
ok I will remove the role? variable since I have connected applications to a division that itself connects to officers so automatically, we already have the officers who can edit that specific application.